sw/host/cryptotest/testvectors/parsers/nist_cavp_hmac_parser.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Parser for converting NIST CAVP Digital Signatures test vectors to JSON.
+
+"""
+# TODO: Add more detailed docstring
+
+import argparse
+import sys
+import json
+import jsonschema
+
+from cryptotest_util import parse_rsp
+
+# NIST vectors allow differently-sized MACs for each hash function
+# (https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Algorithm-Validation-Program/documents/mac/HMACVS.pdf).
+# However, the test vectors only include the longest length for each
+# hash (i.e. the entire hash is the MAC).
+LENGTH_TO_HASH = {
+    "20": "sha1",
+    "28": "sha224",
+    "32": "sha256",
+    "48": "sha384",
+    "64": "sha512",
+}
+
+
+def parse_testcases(args) -> None:
+    raw_testcases = parse_rsp(args.src)
+    test_cases = list()
+
+    # NIST splits the rsp files into sections based on tag length
+    # (which implies which hash function should be used)
+    for section_name in raw_testcases.keys():
+        _, mac_len = section_name.split("=")
+        hash_alg = LENGTH_TO_HASH[mac_len]
+        for test_vec in raw_testcases[section_name]:
+            test_case = {
+                "vendor": "nist",
+                "test_case_id": test_vec["Count"],
+                "algorithm": "hmac",
+                "hash_alg": hash_alg,
+                "key": list(bytes.fromhex(test_vec["Key"])),
+                "message": list(bytes.fromhex(test_vec["Msg"])),
+                "tag": list(bytes.fromhex(test_vec["Mac"])),
+                # All NIST HMAC vectors are expected to pass
+                "result": True,
```

Comment:
Quick question about the API: are these test cases for the hash computation or for validation? If we're just computing the hash and making sure it matches the expected, then we don't need the `result` field right? Unless there are test vectors that cause the hash computation to fail.

Reply:
It's the latter. All the NIST vectors have an expected tag that the HMAC output is supposed to match. However, we need a `result` field because the Wycheproof vectors have some expected failures (see #21149).